### ISSUE TITLE
mismatch in target_link_uri in launch data

### ISSUE THREAD

**Issue by hmoffatt** [celtic-project/wordpress-lti-platform#6]
I'm trying to use deep linking to connect to my tool. When I create the link, the content selection processes launches ok and the link is added to the post:

```
[lti-platform tool=mytool id=aelfw5xp title="Tool name" url=https://mysite.com/launch?app=myapp]Deep link[/lti-platform]
```

Note that the URL for my tool contains a query part which results from the content selection process.

When I try to open the link, the initial login POST has the correct `target_link_uri`:

```
target_link_uri => https://mysite.com/launch?app=myapp
```

but then the id_token has the original URL instead:

```
"https://purl.imsglobal.org/spec/lti/claim/target_link_uri": "https://mysite.com/launch",
```

My tool complains about the mismatch. For all the other LTI platforms I have tested (Moodle/Blackboard/Canvas/Brightspace and others) the id_token claim matches the login data.

**Comment by spvickers** [comment]
Thanks for reporting this; I have been able to replicate a behaviour which appears to be a bug with not properly using the deep linking URL when it is different from the message URL.  I will commit a fix over the next couple of days.

**Comment by spvickers** [comment]
This issue should now be fixed in version 2.2.1

**Comment by hmoffatt** [comment]
Thanks, it's working now.
